### PR TITLE
[ELY-1398] Add support for public API compatibility check

### PR DESCRIPTION
**Pull request by mchoma**
https://issues.jboss.org/browse/ELY-1398

Using SigTest tool [1] two profiles were created
1. to generate method signature file *.sigfile into local maven
repository
2. and check against specified version releasedVersion

Usage:
mvn install -PmethodSignatureGenerate -DskipTests
mvn install -PmethodSignatureCheck -DskipTests

This expects during Elytron release process *.sigfile from local maven
repository will be deployed into central maven repository.

[1] http://wiki.netbeans.org/SigTest

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -88,6 +88,9 @@
         <!-- Modularized JDK support (various workarounds) - activated via profile -->
         <modular.jdk.args/>
         <modular.jdk.props/>
+        <!-- method signature check -->
+        <!-- omitted org.wildfly.security.auth.server.event - breaks generation ELY-1453 -->
+        <public.packages>org.wildfly.security,org.wildfly.security.auth,org.wildfly.security.auth.callback,org.wildfly.security.auth.client,org.wildfly.security.auth.permission,org.wildfly.security.auth.principal,org.wildfly.security.auth.server,org.wildfly.security.auth.util,org.wildfly.security.authz,org.wildfly.security.credential,org.wildfly.security.credential.source,org.wildfly.security.credential.store,org.wildfly.security.evidence,org.wildfly.security.http,org.wildfly.security.key,org.wildfly.security.manager,org.wildfly.security.manager.action,org.wildfly.security.mechanism,org.wildfly.security.password,org.wildfly.security.password.interfaces,org.wildfly.security.password.spec,org.wildfly.security.permission,org.wildfly.security.sasl.util,org.wildfly.security.ssl,org.wildfly.security.auth.server.event</public.packages>
```
Comment (fjuma) [inline pom.xml]: Looks like org.wildfly.security.auth.server.event hasn't been omitted, it's still present at the end of the packages.

File: pom.xml [hunk 2]
```diff
@@ -542,6 +545,53 @@
...
+                        <executions>
+                            <execution>
+                                <goals>
+                                    <goal>check</goal>
+                                </goals>
+                            </execution>
+                        </executions>
+                        <configuration>
+                            <packages>${public.packages}</packages>
+                            <releaseVersion>1.1.8.Final</releaseVersion>
```
Comment (fjuma) [inline pom.xml]: Is this meant to be the Elytron version? Could ${project.version} be used here instead?

Comment (mchoma) [reply pom.xml]: It is meant to be Elytron version, but some important version in history. In this case latest 1.1.x release which contains generated method signature file. There is no such release now. This is just guess.

${project.version} will resolve to something like 1.2.x.Final it is not what we want here.

File: pom.xml (hunk 1 above)
Comment (mchoma) [reply pom.xml]: Good catch. Will fix.

File: pom.xml (hunk 2 above)
Comment (honza889) [reply pom.xml]: +1

Comment (fjuma) [reply pom.xml]: Makes sense. Just one more minor nit, the commit should reference an ELY issue.

Comment (honza889) [reply pom.xml]: Looks like ELY-1398 - please reference it in commit message and in pull request title ;)
https://issues.jboss.org/browse/ELY-1398

**Comment by honza889** [comment]
Looks ok, list of public packages is complete (except mentioned ELY-1453).
But maybe we should wait with merging until releaseVersion against which we will test will be decided?

**Comment by honza889** [comment]
btw works well - just successfully used for https://github.com/wildfly-security/wildfly-elytron/pull/1039 fix verification 👍

File: pom.xml [hunk 3]
```diff
@@ -88,6 +88,9 @@
         <!-- Modularized JDK support (various workarounds) - activated via profile -->
         <modular.jdk.args/>
         <modular.jdk.props/>
+        <!-- method signature check -->
+        <!-- omitted org.wildfly.security.auth.server.event - breaks generation ELY-1453 -->
+        <public.packages>org.wildfly.security,org.wildfly.security.auth,org.wildfly.security.auth.callback,org.wildfly.security.auth.client,org.wildfly.security.auth.permission,org.wildfly.security.auth.principal,org.wildfly.security.auth.server,org.wildfly.security.auth.util,org.wildfly.security.authz,org.wildfly.security.credential,org.wildfly.security.credential.source,org.wildfly.security.credential.store,org.wildfly.security.evidence,org.wildfly.security.http,org.wildfly.security.key,org.wildfly.security.manager,org.wildfly.security.manager.action,org.wildfly.security.mechanism,org.wildfly.security.password,org.wildfly.security.password.interfaces,org.wildfly.security.password.spec,org.wildfly.security.permission,org.wildfly.security.sasl.util,org.wildfly.security.ssl</public.packages>
```
Comment (fjuma) [inline pom.xml]: Now that https://github.com/wildfly-security/wildfly-elytron/pull/1041 has been merged, the following packages should also be added to this list:

* org.wildfly.security.asn1
* org.wildfly.security.x500
* org.wildfly.security.x500.cert

**Comment by mchoma** [comment]
Somehow managed to close this PR by mistake with git command. 
Follow up https://github.com/wildfly-security/wildfly-elytron/pull/1066

@hkalina please +1 if you didn't change your mind :)

**Pull request closed without merging**